src/common/util/serialization.js

```diff
@@ -45,16 +46,47 @@ define([
      * @returns {Promise}
      */
     exports.exportProjectToFile = function exportProjectToFile(project, blobClient, parameters, callback) {
-        var fileName;
-
-        return storageUtils.getProjectJson(project, {
-            branchName: parameters.branchName,
-            commitHash: parameters.commitHash,
-            rootHash: parameters.rootHash,
-            tagName: parameters.tagName,
-            kind: parameters.kind
-        })
+        let fileName;
+        let libInfo;
+
+        return storageUtils.getRootHash(project, parameters)
+            .then(function (rootHash) {
+                if (!parameters.libraryName) {
+                    return {
+                        branchName: parameters.branchName,
+                        commitHash: parameters.commitHash,
+                        rootHash: parameters.rootHash,
+                        tagName: parameters.tagName,
+                        kind: parameters.kind
+                    };
+                }
+
+                const core = new Core(project, { globConf: project.gmeConfig, logger: project.logger });
+
+                return core.loadRoot(rootHash)
+                    .then(function (rootNode) {
+                        const libNode = core.getLibraryRoot(rootNode, parameters.libraryName);
+                        libInfo = core.getLibraryInfo(rootNode, parameters.libraryName);
+                        return { rootHash: core.getHash(libNode) };
+                    });
+            })
+            .then(function (exportContext) {
+                return storageUtils.getProjectJson(project, {
+                    branchName: exportContext.branchName,
+                    commitHash: exportContext.commitHash,
+                    rootHash: exportContext.rootHash,
+                    tagName: exportContext.tagName,
+                    kind: exportContext.kind
+                });
+            })
             .then(function (rawJson) {
+                if (parameters.libraryName) {
+                    // We need to adjust this when the exportee was a library.
+                    delete rawJson.kind; // This is lost at this point.
```

Comment:
i believe i know where this come from, but is this really the good behavior?
shouldn't we enhance the library info with the kind to begin with?